http.js

```diff
@@ -0,0 +1,254 @@
+    })
+  })
+
+  // Source maps. Each source map is stored as 'foo.js.map' within their
+  // respective node. So in order to figure out the right osurce map we must
+  // derive figure out where the extension comes from.
+  router.route(/\/([a-zA-Z0-9-_.]+)\.map$/, function (req, res, params) {
+    var source = params[1]
+    var ext = path.extname(source.replace(/\.map$/, ''))
+    var type = source === 'bankai-reload.js'
+      ? 'reload'
+      : source === 'bankai-service-worker.js'
+        ? 'service-worker'
+        : ext === '.js'
+          ? 'scripts'
+          : 'unknown'
+    compiler.sourceMaps(type, source, function (err, node) {
+      if (err) {
+        res.statusCode = 404
+        return res.end(err.message)
+      }
+      res.setHeader('content-type', 'application/json')
+      gzip(node.buffer, req, res)
+    })
+  })
+
+  router.route(/^\/assets\/(.*)$/, function (req, res, params) {
+    var prefix = 'assets' // TODO: also accept 'content'
+    var name = prefix + '/' + params[1]
+    compiler.assets(name, function (err, node) {
+      if (err) {
+        res.statusCode = 404
+        return res.end(err.message)
+      }
+      res.end(node.buffer)
+    })
+  })
+
+  router.route(/\/reload/, function sse (req, res) {
+    var connected = true
+    emitter.on('scripts:bundle', reloadScript)
+    emitter.on('style:bundle', reloadStyle)
+    state.sse += 1
+    if (!quiet) render()
+
+    res.writeHead(200, {
+      'Content-Type': 'text/event-stream',
+      'Cache-Control': 'no-cache'
+    })
+    res.write('retry: 10000\n')
+
+    var interval = setInterval(function () {
+      res.write(`id:${id++}\ndata:{ "type:": "heartbeat" }\n\n`)
+    }, 4000)
+
+    // Attach an error handler, but no need to actually handle the error.
+    // This is a bug in Node core according to mcollina which will be fixed
+    // in a future Node release. Let's keep this in place as long as v8.x.x of
+    // Node isn't in LTS yet.
+    res.on('error', disconnect)
+    req.on('error', disconnect)
+
+    req.connection.addListener('close', disconnect, false)
+
+    function disconnect () {
+      clearInterval(interval)
+      if (connected) {
+        emitter.removeListener('scripts:bundle', reloadScript)
+        emitter.removeListener('style:bundle', reloadStyle)
+        connected = false
+        state.sse -= 1
+        if (!quiet) render()
+      }
+    }
+
+    function reloadScript (node) {
+      var msg = JSON.stringify({ type: 'scripts' })
+      res.write(`id:${id++}\ndata:${msg}\n\n`)
+    }
+
+    function reloadStyle (node) {
+      var msg = JSON.stringify({
+        type: 'style',
+        bundle: node.buffer.toString()
+      })
+      res.write(`id:${id++}\ndata:${msg}\n\n`)
+    }
+  })
+
+  router.default(function (req, res, next) {
+    var url = req.url
+    compiler.documents(url, function (err, node) {
+      if (err) {
+        return compiler.documents('/404', function (err, node) {
```

Comment:
i was just looking this over trying to implement a default: you've got two `err`, the other one should probably be `error` or something

Reply:
o/ Thanks for reporting - this is fine tho! The closures work alright in this case; there's no bugs here (: